### PR TITLE
chore: convert payment request CTA to select

### LINKED ISSUE #6470: Improve design of express checkout button settings
### Description
In a user flow review, the group called out the lack of hierarchy in the express checkout button settings:
image

The above screenshot was taken on `wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments&method=woopay`

We would like to see more hierarchy and visual separation between sections.

cc @elizaan36  @kellychoffman  @pierorocca

### PR DESCRIPTION
**Pull request by frosso**
Fixes https://github.com/Automattic/woocommerce-payments/issues/6470

#### Changes proposed in this Pull Request

Converting the CTA on the payment request & WooPay settings sub-pages to a "Select". See https://github.com/Automattic/woocommerce-payments/issues/6470#issuecomment-1597718155 for mockup.
I also slightly tweaked the margin-bottom below the options, to make it match the designs a little closer.

#### Testing instructions

- Go to http://wcpay.test/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments&method=woopay
- The option for the CTA should be updated
- Go to http://wcpay.test/wp-admin/admin.php?page=wc-settings&tab=checkout&section=woocommerce_payments&method=payment_request
- Similar behavior as above

Screenshot 2024-02-08 at 2 52 26 PM
Screenshot 2024-02-08 at 2 52 22 PM

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8155` or branch name `chore/convert-payment-request-cta-to-select` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=chore/convert-payment-request-cta-to-select) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 45a2d63bb83351510e05937f4d5b798e19eda6c2
- Build time: 2024-02-13 16:24:40 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/settings/card-body/styles.scss [hunk 1]
```diff
@@ -39,7 +39,7 @@
 
 	// spacing in the "Express checkouts" settings page
 	.components-radio-control__option {
-		margin-bottom: 18px;
+		margin-bottom: 10px;
```
Comment (frosso) [inline client/settings/card-body/styles.scss:47]: I updated this value to slightly better match the designs

File: client/settings/express-checkout-settings/test/payment-request-button-preview.test.js [hunk 2]
```diff
@@ -39,6 +39,19 @@ jest.mock( 'wcpay/data', () => {
 	};
 } );
 
+const render = ( ui, options ) =>
+	baseRender( ui, {
+		wrapper: ( { children } ) => (
+			<div
+				id="wcpay-express-checkout-settings-container"
```
Comment (frosso) [inline client/settings/express-checkout-settings/test/payment-request-button-preview.test.js:46]: This element is rendered here ( https://github.com/Automattic/woocommerce-payments/blob/a920abdca8624cd96174c1aeba277835e6cdd564/includes/class-wc-payment-gateway-wcpay.php#L883-L883 ), and I needed to add this wrapper to more closely match the behavior of the components when in a non-test environment.

**Review by mdmoore — CHANGES_REQUESTED** [review]
Nice one! I just noticed that the core select component is setting a `max-width: 25rem` that's preventing the dropdown from working when trying to click on it outside of that width, such as on the down arrow.

Screenshot 2024-02-12 at 10 39 28 AM

Otherwise, it's working well!

**Comment by frosso** [comment]
@mdmoore good eyes, thanks for pointing that out! I'll take a look momentarily by my EOD.

**Review by mdmoore — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**